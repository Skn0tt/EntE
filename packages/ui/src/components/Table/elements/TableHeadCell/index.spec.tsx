import * as React from "react";
import { TableHeadCell } from "./index";
import { shallow } from "enzyme";

describe("TableHeadCell", () => {
  const onClick = jest.fn();

  const comp = shallow(
    <TableHeadCell
      onClick={onClick}
      classes={{}}
      sortUp
      active
      tooltip="I bims"
    />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });

  it("calls onClick when clicking", () => {
    comp.find(".tableSortLabel").simulate("click");
    expect(onClick).toBeCalled();
  });
});
