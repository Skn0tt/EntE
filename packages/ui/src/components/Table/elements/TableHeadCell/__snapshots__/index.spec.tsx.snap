// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadCell renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    active={true}
    classes={Object {}}
    onClick={[MockFunction]}
    sortUp={true}
    tooltip="I bims"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Tooltip)
        enterDelay={300}
        title="I bims"
      >
        <WithStyles(TableSortLabel)
          active={true}
          className="tableSortLabel"
          direction="asc"
          onClick={[MockFunction]}
        />
      </WithStyles(Tooltip)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(TableSortLabel)
          active={true}
          className="tableSortLabel"
          direction="asc"
          onClick={[MockFunction]}
        />,
        "enterDelay": 300,
        "title": "I bims",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "children": undefined,
          "className": "tableSortLabel",
          "direction": "asc",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Tooltip)
          enterDelay={300}
          title="I bims"
        >
          <WithStyles(TableSortLabel)
            active={true}
            className="tableSortLabel"
            direction="asc"
            onClick={[MockFunction]}
          />
        </WithStyles(Tooltip)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(TableSortLabel)
            active={true}
            className="tableSortLabel"
            direction="asc"
            onClick={[MockFunction]}
          />,
          "enterDelay": 300,
          "title": "I bims",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": undefined,
            "className": "tableSortLabel",
            "direction": "asc",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
